### PR TITLE
[JSON-RPC] Fixing flaky tests

### PR DESCRIPTION
**Pull request by AnomalRoil**
This is avoiding the edge case where the diem_ledger_version would advance between the get_metadata and the get_state_proof calls in the integration test and cause the assert to crash since the hashes has more than 1 transaction but less than 10, since the diem_ledger_version probably won't advance 9 versions at once...

It is also bringing back the integration test for get_events_with_proof

### CODE REVIEW

**Comment by libra-action** [comment]
**This PR may have modified JSON-RPC server code.**

Breaking changes policy:

1. Chaning JSON-RPC method signatures or removing/modifying fields
   in the response in /V1 will not be accepted.
2. Adding fields is ok.
3. Adding a new JSON-RPC method is ok.

Please ensure you have also done the following to update the docs:

1. Update json-rpc/API-CHANGELOG.md.
2. Add/update type/method documentation under json-rpc/docs.

File: json-rpc/tests/integration_test.rs [hunk 1]
```diff
@@ -999,6 +1005,51 @@ fn create_test_cases() -> Vec<Test> {
...
+                    let hash = event.event.hash();
+
+                    // We verify the proof of the event
+                    assert!(event.proof.verify(ledger_info, hash, event.transaction_version, event.event_index).is_ok());
+
+                    // We can now use our verified events
+                    events.push((event.transaction_version, event.event).try_into().unwrap());
+                }
+
+                assert!(events.len()==3);
```
Comment (xli) [inline json-rpc/tests/integration_test.rs:1050]: assert_eq!(events.len(), 3)

**Review by xli — APPROVED** [review]

**Comment by AnomalRoil** [comment]
What's up with the failing tests @xli?
I rebased to make sure but it didn't help. Can we still land?
